json_serialization/format.nim

```diff
@@ -8,3 +8,32 @@ template supports*(_: type Json, T: type): bool =
   # The JSON format should support every type
   true
 
+template flavorUsesAutomaticObjectSerialization*(T: type DefaultFlavor): bool = true
+template flavorOmitsOptionalFields*(T: type DefaultFlavor): bool = false
+template flavorRequiresAllFields*(T: type DefaultFlavor): bool = false
+template flavorAllowsUnknownFields*(T: type DefaultFlavor): bool = false
+
+# We create overloads of these traits to force the mixin treatment of the symbols
+type DummyFlavor* = object
+template flavorUsesAutomaticObjectSerialization*(T: type DummyFlavor): bool = true
+template flavorOmitsOptionalFields*(T: type DummyFlavor): bool = false
+template flavorRequiresAllFields*(T: type DummyFlavor): bool = false
+template flavorAllowsUnknownFields*(T: type DummyFlavor): bool = false
+
+template createJsonFlavor*(FlavorName: untyped,
+                           mimeTypeValue = "application/json",
+                           automaticObjectSerialization = false,
```

Comment:
this should probably be `true` since json is typically used for interop via standards - the way to get there would be to have a deprecated overload that has the old behavior

Reply:
I don't understand the remark. The `false` value here means that you'll get the explicit override behavior by default.

Reply:
ah oops, I read it wrong. :+1: